### PR TITLE
Add signMessage and verifyMessage functions

### PR DESCRIPTION
**Pull request by intelliot**
- Sign an arbitrary string with a private key
- Verify a signature with a public key

### CODE REVIEW

File: src/offline/sign-message.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+const binary = require('ripple-binary-codec')
+const {validate} = common
+
+function signMessage(message: string,
+  privateKey: string
+): string {
+  validate.signMessage({message, privateKey})
+
+  const signingData = binary.encode({
+    message: message
```
Comment (sublimator) [inline src/offline/sign-message.js]: This isn't going to work: `binary.encode` expects ledger entry, or transaction json, as spoken by rippled. It silently discards fields, so you're probably turning the message into a signingData of empty bytes.

Comment (sublimator) [reply src/offline/sign-message.js]: https://github.com/ripple/ripple-binary-codec/issues/10

Comment (sublimator) [reply src/offline/sign-message.js]: You probably just want to encode the message as utf-8 ?

Comment (sublimator) [reply src/offline/sign-message.js]: With some kind of `Hash Prefix` ?

Comment (intelliot) [reply src/offline/sign-message.js]: Thank you!

Not sure how to select a `Hash Prefix`. Perhaps just encoding as utf-8 and using its hex representation is sufficient?

Comment (sublimator) [reply src/offline/sign-message.js]: @JoelKatz is the chief crypto guy @ ripple :)

Tell him your use case.

Comment (sublimator) [reply src/offline/sign-message.js]: > Perhaps just encoding as utf-8 and using its hex representation is sufficient?

Perhaps, but I don't know what you're doing, and I'm not a crypto guy. 

Ask @justmoon or @JoelKatz

**Pull request closed without merging**